asmcomp/linearize.ml

```diff
@@ -311,7 +311,7 @@ let rec linear i n =
 let fundecl f =
   { fun_name = f.Mach.fun_name;
     fun_body = linear f.Mach.fun_body end_instr;
-    fun_fast = f.Mach.fun_fast;
+    fun_fast = List.mem Cmm.Reduce_code_size f.Mach.fun_codegen_options;
     fun_dbg  = f.Mach.fun_dbg;
```

Comment:
Here you're reversing the logic again.  "Reduce code size" means "fun_fast = false", not "fun_fast = true".